src/Sulu/Bundle/SecurityBundle/Resources/config/services.xml

```diff
@@ -146,7 +140,9 @@
         </service>
 
         <service id="sulu_security.user_repository" class="%sulu_security.user_repository.class%">
-            <factory service="sulu_security.user_repository_factory" method="getRepository"/>
+            <factory service="doctrine.orm.entity_manager" method="getRepository"/>
+
+            <argument>SuluSecurityBundle:User</argument>
```

Comment:
use the container parameter for that. otherwise it wouldn't work for overridden users.